woocommerce/woocommerce
# Added tests for order search

### All Submissions:

### Changes proposed in this Pull Request:



Closes #30673.

### How to test the changes in this Pull Request:

1. Clone this repository and run the setup scripts for the e2e environment.
2. Run the API tests for Orders: `npm test -- --group=orders`.
3. The `Order Search API tests` should pass like so:
```
 PASS  tests/orders/order-search.test.js
  Order Search API tests
    ✓ can search by orderId (88ms)
    ✓ can search by billing first name (81ms)
    ✓ can search by billing last name (79ms)
    ✓ can search by billing company name (76ms)
    ✓ can search by billing address 1 (74ms)
    ✓ can search by billing address 2 (76ms)
    ✓ can search by billing city name (79ms)
    ✓ can search by billing post code (78ms)
    ✓ can search by billing email (85ms)
    ✓ can search by billing phone (81ms)
    ✓ can search by billing state (75ms)
    ✓ can search by shipping first name (79ms)
    ✓ can search by shipping last name (76ms)
    ✓ can search by shipping address 1 (75ms)
    ✓ can search by shipping address 2 (79ms)
    ✓ can search by shipping city (90ms)
    ✓ can search by shipping post code (85ms)
    ✓ can search by shipping state (117ms)
    ✓ can search by item name (98ms)
    ✓ can return an empty result set (95ms)

Test Suites: 1 passed, 1 of 5 total
Tests:       20 passed, 20 total
```
4. Verify through the UI or the "List all orders" endpoint that no orders were left over by the test.

### Other information:


### Changelog entry

> Added: Order Search API tests

### FOR PR REVIEWER ONLY:
